src/select.ts

```diff
@@ -18,22 +17,23 @@ export function Select<TState = any, TValue = any>(
   ...paths: string[]
 ): (target: any, name: string) => void;
 export function Select<TState = any, TValue = any>(
-  selectorOrFeature: string | Selector<TState, TValue>,
+  selectorOrFeature?: string | Selector<TState, TValue>,
   ...paths: string[]
 ) {
   return function(target: any, name: string): void {
     let fn: Selector<TState, TValue>;
     // Nothing here? Use propery name as selector
-    if (typeof selectorOrFeature === 'undefined') {
+    if (!selectorOrFeature) {
       selectorOrFeature = name;
     }
     // Handle string vs Selector<TState, TValue>
     if (typeof selectorOrFeature === 'string') {
-      if (paths.length) {
-        selectorOrFeature = [selectorOrFeature, ...paths].join('.');
+      let propsArray = [selectorOrFeature, ...paths];
+      if (!paths.length) {
+        propsArray = selectorOrFeature.split('.');
       }
-      fn = fastPropGetter(selectorOrFeature);
-    } else {
+      fn = fastPropGetter(propsArray);
+    } else if (typeof selectorOrFeature === 'function') {
```

Comment:
I don't think we need this check: if it's not a `string` than it's a `Selector<TState, TValue>`.